src/pages/Signup/index.jsx

```diff
@@ -57,7 +57,7 @@ const SignupScreen = () => {
     if (!regex.test(name)) {
       setUsernameAvailable(false);
     } else {
-      debounce(findUsernameInDB);
+      debounce(findUsernameInDB());
```

Comment:
aren't we passing function reference to the debounce function instead of actually calling the function?